src/ICSharpCode.SharpZipLib/Zip/ZipOutputStream.cs

```diff
@@ -327,7 +327,14 @@ public void PutNextEntry(ZipEntry entry)
 				}
 				else
 				{
-					WriteLeInt(entry.IsCrypted ? (int)entry.CompressedSize + ZipConstants.CryptoHeaderSize : (int)entry.CompressedSize);
+					int entryCompressedSize = (int)entry.CompressedSize;
+
+					if (entry.IsCrypted)
+					{
+						entryCompressedSize += entry.EncryptionOverheadSize;
+					}
+
+					WriteLeInt(entryCompressedSize);
```

Comment:
Why check IsCrypted twice?
```suggestion
					WriteLeInt((int)entry.CompressedSize + entry.EncryptionOverheadSize);
```